# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../lualdap/lualdap-1.1.0-cpp-define-LDAP_DEPRECATED.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 796f32bb5afd8efd316f3829d7e0956a976a1aef Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Wed, 5 Sep 2012 16:12:09 +0200
Subject: [PATCH] build: fixed to define LDAP_DEPRECATED pre-processor flag

This is fixing minor build warings about implicit declarations because
lualdap is using deprecated ldap functions, which are only defined in
the header if LDAP_DEPRECATED is defined.
---
 Makefile |    2 +-
 config   |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 6a4f06d..0ffc48c 100755
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ OBJS= src/lualdap.o $(COMPAT_O)
 
 
 src/$(LIBNAME): $(OBJS)
-	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/$(LIBNAME) $(OBJS) $(OPENLDAP_LIB)
+	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CPPFLAGS) $(CFLAGS) $(LIB_OPTION) -o src/$(LIBNAME) $(OBJS) $(OPENLDAP_LIB)
 
 $(COMPAT_DIR)/compat-5.1.o: $(COMPAT_DIR)/compat-5.1.c
 	$(CC) -c $(CFLAGS) -o $@ $(COMPAT_DIR)/compat-5.1.c
diff --git a/config b/config
index 84543b3..c3404fa 100755
--- a/config
+++ b/config
@@ -21,6 +21,7 @@ COMPAT_DIR= ../compat/src
 WARN= -O2 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -ansi
 INCS= -I$(LUA_INC) -I$(OPENLDAP_INC) -I$(COMPAT_DIR)
 CFLAGS= $(WARN) $(INCS)
+CPPFLAGS= -DLDAP_DEPRECATED
 CC= gcc
 
 # $Id: config,v 1.5 2006-07-24 01:42:06 tomas Exp $
-- 
1.6.6.2

